internal/buildsign/ocpbuild/ocpbuildmanager.go

```diff
@@ -0,0 +1,173 @@
+}
+
+type ocpbuildManagerImpl struct {
+	client client.Client
+}
+
+func newOCPBuildManager(client client.Client) ocpbuildManager {
+	return &ocpbuildManagerImpl{
+		client: client,
+	}
+}
+
+func (omi *ocpbuildManagerImpl) isOCPBuildChanged(existingBuild *buildv1.Build, newBuild *buildv1.Build) (bool, error) {
+	existingAnnotations := existingBuild.GetAnnotations()
+	newAnnotations := newBuild.GetAnnotations()
+	if existingAnnotations == nil {
+		return false, fmt.Errorf("annotations are not present in the existing build %s", existingBuild.Name)
+	}
+	if existingAnnotations[hashAnnotation] == newAnnotations[hashAnnotation] {
+		return false, nil
+	}
+	return true, nil
+}
+
+func (omi *ocpbuildManagerImpl) getModuleOCPBuildByKernel(ctx context.Context,
+	modName,
+	modNamespace,
+	kernelVersion,
+	ocbuildType string,
+	owner metav1.Object) (*buildv1.Build, error) {
+	buildList := buildv1.BuildList{}
+
+	opts := []client.ListOption{
+		client.MatchingLabels(omi.ocpbuildLabels(modName, kernelVersion, ocbuildType)),
+		client.InNamespace(modNamespace),
+	}
+
+	if err := omi.client.List(ctx, &buildList, opts...); err != nil {
+		return nil, fmt.Errorf("could not list Build: %v", err)
+	}
+
+	// filter OCP builds by owner, since they could have been created by the preflight
+	// when checking that specific module
+	moduleOwnedOCPBuilds := filterOCPBuildsByOwner(buildList.Items, owner)
+
+	if n := len(moduleOwnedOCPBuilds); n == 0 {
+		return nil, ErrNoMatchingBuild
+	} else if n > 1 {
+		return nil, fmt.Errorf("expected 0 or 1 Builds, got %d", n)
+	}
+
+	return &moduleOwnedOCPBuilds[0], nil
+}
+
+func (omi *ocpbuildManagerImpl) getModuleOCPBuilds(ctx context.Context, modName, modNamespace, ocpbuildType string, owner metav1.Object) ([]buildv1.Build, error) {
+	buildList := buildv1.BuildList{}
+
+	opts := []client.ListOption{
+		client.MatchingLabels(moduleLabels(modName, ocpbuildType)),
+		client.InNamespace(modNamespace),
+	}
+
+	if err := omi.client.List(ctx, &buildList, opts...); err != nil {
+		return nil, fmt.Errorf("could not list Build: %v", err)
+	}
+
+	// filter OCP builds by owner, since they could have been created by the preflight
+	// when checking that specific module
+	moduleOwnedBuilds := filterOCPBuildsByOwner(buildList.Items, owner)
+
+	return moduleOwnedBuilds, nil
+}
+
+func (omi *ocpbuildManagerImpl) getOCPBuildStatus(build *buildv1.Build) (Status, error) {
+	switch build.Status.Phase {
+	case buildv1.BuildPhaseComplete:
+		return StatusCompleted, nil
+	case buildv1.BuildPhaseRunning, buildv1.BuildPhasePending:
+		return StatusInProgress, nil
+	case buildv1.BuildPhaseFailed, buildv1.BuildPhaseCancelled, buildv1.BuildPhaseError:
+		return StatusFailed, nil
+	default:
+		return "", fmt.Errorf("unknown status: %v", build.Status.Phase)
+	}
+}
+
+func (omi *ocpbuildManagerImpl) createOCPBuild(ctx context.Context, build *buildv1.Build) error {
+	err := omi.client.Create(ctx, build)
+	if err != nil {
+		return err
+	}
+	return nil
+}
```

Comment:
Do we need this method? calling it or calling directly `client.Create` will look the same.